src/Microsoft.ML/Data/CollectionDataSource.cs

```diff
@@ -19,17 +20,23 @@ public static class CollectionDataSource
         /// <summary>
         /// Creates pipeline data source. Support shuffle.
         /// </summary>
-        public static ILearningPipelineLoader Create<T>(IList<T> data) where T : class
+        public static ILearningPipelineLoader Create<T>(IList<T> data,
+            Dictionary<string, int[]> vectorSizes = null, bool inferVectorSizesFromCollection = false) where T : class
```

Comment:
>bool inferVectorSizesFromCollection = false [](start = 58, length = 43)

Considering fact what I need to pass vector size to pipeline during training as well, and during that time I don't have collection it's useless here, but I can move it to deeper level of ComponentCreation.CreateDataView if we plan to use it after API overhaul.